tensorflow/python/framework/test_util.py

```diff
@@ -266,6 +266,13 @@ def setUp(self):
     self._ClearCachedSession()
     random.seed(random_seed.DEFAULT_GRAPH_SEED)
     np.random.seed(random_seed.DEFAULT_GRAPH_SEED)
+    # Note: The following line is necessary because some test methods may error
+    # out from within nested graph contexts (e.g., via assertRaises and
+    # assertRaisesRegexp), which may leave ops._default_graph_stack non-empty
+    # under certain versions of Python. That would cause
+    # ops.reset_default_graph() to throw an exception if the stack were not
+    # cleared first.
```

Comment:
Good question! This seems strange to me, too. But this is what happens in Mac+Python3.5: see code at: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/client/session_test.py#L1556

This code calls "with sess" twice. By using print debugging, I can see that during the first call, which is legitimate, a Graph object is pushed to the default graph stack and a Session object is pushed to the default session stack. But during the second call, which is illegal and triggers an exception about re-entry, the finally block of _only_ the default session stack is called. That of the default graph stack is not called. This is unique to Mac+Python3.5. In other systems and python versions, the finally blocks of both the default session stack and the default graph stack get called.

Any thoughts?